vmm_core/virt_support_x86emu/src/emulate.rs

```diff
@@ -801,6 +805,47 @@ impl<T: EmulatorSupport, U> EmulatorCpu<'_, T, U> {
                 },
             })
     }
+
+    pub fn handle_guest_memory_error(
+        &mut self,
+        gva: u64,
+        gpa: u64,
+        err: &GuestMemoryError,
+        user_mode: bool,
+    ) -> bool {
+        match err.kind() {
+            guestmem::GuestMemoryErrorKind::VtlProtected => {
+                // TODO: support for this type for non-hardware-CVMs
```

Comment:
The way I imagine this working is that the `page_fault` impl for the non-CVM memory backing will query the hypervisor for the page.